spec/requests/publishers/ats_api/v1/update_vacancy_service_spec.rb

```diff
@@ -0,0 +1,74 @@
+require "rails_helper"
+
+RSpec.describe Publishers::AtsApi::V1::UpdateVacancyService do
+  subject(:service) { described_class.new(vacancy, params) }
+
+  let(:vacancy) { create(:vacancy, :external, organisations: [school]) }
+  let(:school) { create(:school) }
+  let(:school_urns) { { school_urns: [school.urn] } }
+  let(:job_title) { vacancy.job_title }
+  let(:job_advert) { vacancy.job_advert }
+  let(:job_roles) { vacancy.job_roles }
+  let(:working_patterns) { vacancy.working_patterns }
+  let(:params) do
+    {
+      external_reference: "new-ref",
+      job_title: job_title,
+      job_advert: job_advert,
+      external_advert_url: vacancy.external_advert_url,
+      job_roles: job_roles,
+      contract_type: vacancy.contract_type,
+      phases: vacancy.phases,
+      working_patterns: working_patterns,
+      expires_at: vacancy.expires_at,
+      skills_and_experience: vacancy.skills_and_experience,
+      salary: vacancy.salary,
+      schools: school_urns,
+    }
+  end
+
+  describe "#call" do
+    context "when the update is successful" do
+      it "updates only the attributes that differ from the original vacancy" do
+        service.call
+        vacancy.reload
+
+        expect(vacancy.working_patterns).to eq(%w[full_time])
+        expect(vacancy.external_reference).to eq("new-ref")
+      end
+    end
+
+    context "when organisations are invalid" do
+      let(:school_urns) { { school_urns: [9999] } }
+
+      it "raises ActiveRecord::RecordNotFound" do
+        expect { service.call }.to raise_error(ActiveRecord::RecordNotFound, "No valid organisations found")
+      end
+    end
+
+    context "when the vacancy fails validation" do
```

Comment:
This feels strange - effectively the service is the controller. Would this code not be better in the controller?

Reply:
I think it does a little too much - returning controller statuses like 422 look like they belong in the controller

Reply:
The main issue I see here is that the Service Object lives in the controllers folder and is tested as a request. 

Is it a controller or a service object? If it cannot be placed within the service objects folder and unit tested wouldn't that hint that we're dealing with the wrong abstraction?